LiteCore/Query/SQLiteDateTimeHelpers.hh

```diff
@@ -0,0 +1,220 @@
+#pragma once
+
+#include "ParseDate.hh"
+#include "date/date.h"
+#include <chrono>
+#include <cmath>
+#include <ratio>
+#include <cstdint>
+#include "SQLiteFleeceUtil.hh"
+
+using namespace std::chrono;
+using namespace date;
+using namespace fleece;
+
+namespace date {
+
+    using quarters    = duration<int, detail::ratio_multiply<std::ratio<3>, months::period>>;
+    using decades     = duration<int, detail::ratio_multiply<std::ratio<10>, years::period>>;
+    using centuries   = duration<int, detail::ratio_multiply<std::ratio<100>, years::period>>;
+    using millenniums = duration<int, detail::ratio_multiply<std::ratio<1000>, years::period>>;
+
+    typedef struct {
+        int64_t year;
+        int64_t doy;
+        int64_t hour;
+        int64_t minute;
+        int64_t second;
+        int64_t millisecond;
+    } DateDiff;
+
+}  // namespace date
+
+namespace litecore {
+    inline int64_t getQuarter(const DateTime& t) { return (t.M + 2) / 3; }
+
+    inline int64_t leapYearsWithin(int64_t year) {
+        if ( year > 0 ) {
+            year--;
+        } else {
+            year++;
+        }
+
+        return (year / 4) - (year / 100) + (year / 400);
+    }
+
+    inline int64_t leapYearsBetween(const int64_t start, const int64_t end) {
```

Comment:
I'm concerned about why this is necessary....doesn't date.h already take leap years into account with its math operators?